packages/bot/src/struct/CommandHandler.ts

```diff
@@ -165,11 +209,25 @@ export class CommandHandler {
 		const path = join(dirname(fileURLToPath(import.meta.url)), '..', 'commands');
 		const files = readdirRecurse(path, { fileExtensions: ['js'] });
 
+		const subcommands: { rootcmd: string; command: Subcommand }[] = [];
 		for await (const file of files) {
 			const mod = (await import(pathToFileURL(file).toString())) as { default: CommandConstructor };
 			const command = container.resolve(mod.default);
 
-			this.commands.set(command.interactionOptions.name, command);
+			// check if command is a subcommand via seeing if the directory it's in is a subdirectory of the command directory
+			const directory = dirname(file).split('/').pop()!;
+			const isSubcommand = (cmd: Command | Subcommand | CommandWithSubcommands): cmd is Subcommand =>
+				!['commands', 'context-menus'].includes(directory) && !file.endsWith('index.js');
+
+			if (isSubcommand(command)) {
+				subcommands.push({ rootcmd: directory, command });
```

Comment:
No need for that extra array
```suggestion
				this.commands.set(`${directory}-${$command.interactionOptions.name}`, command);
```